src/app/content/utils/canonicalUrl.ts

```diff
@@ -18,13 +19,17 @@ export async function getCanonicalUrlParams(
   ].filter((id) => !!BOOKS[id]);
 
   for (const id of canonicals) {
-    const version = BOOKS[id].defaultVersion;
+    const version = bookVersion || BOOKS[id].defaultVersion;
```

Comment:
you don't want to always use this version, because if you're pointing to another book this version will be invalid. i think you should change the `canonicals` array on line 16 to have uuid AND version, instead of just uuid, that way when you set the default (line 18 `bookId, // use the current book as a last resort`) you'll put the book and id there